.github/workflows/_buildpacks-prepare-release.yml

```diff
@@ -23,30 +33,40 @@ on:
 jobs:
   prepare-release:
     name: Prepare Release
-    runs-on: pub-hk-ubuntu-22.04-small
+    runs-on: ${{ inputs.ip_allowlisted_runner }}
     steps:
+      - name: Get token for GH application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ inputs.app_id }}
+          private_key: ${{ secrets.app_private_key }}
+
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          token: ${{ steps.generate-token.outputs.app_token }}
```

Comment:
The repo is public, so I would have thought the token isn't needed? Or is this as a way to test the token works early, rather than only find out right at the end when the "open a PR" step happens? (Or in the case of dry-run mode, the open a PR step doesn't happen at all, so wouldn't be tested.)

Reply:
oh, so this is actually related to how you can get signed commits for your GH App.  it's possible to do a bunch of the git setup manually but, if provided a token, the checkout command will automatically configure the git client with the required credentials to use for later commits that can be attributed to the bot.